deltaspike/modules/data/api/src/main/java/org/apache/deltaspike/data/api/Query.java

```diff
@@ -83,7 +82,11 @@
     /**
      * For native queries only, whether or not this query returns the defined entity class or not.
      * Due to type erasure from generics, we don't have runtime information about the return collection
+     *
+     * @deprecated Native queries can deduce the correct return type and mapping. This property has no effect and can be
+     *             removed safely.
      */
+    @Deprecated
     boolean returnsEntity() default true;
```

Comment:
Please delete this parameter, just like in your original PR.  This was only added this week to address this problem originally.